File: FDK2R/FdkRLib/Fdk2R/RSoftFxHost/FdkSessionInfo.cs
using System;
using log4net;
using SoftFX.Extended;

namespace RHost
{
	public static class FdkSessionInfo
	{
		private static DataFeed Feed
		{
			get { return FdkHelper.Wrapper.ConnectLogic.Feed; }
		}

		public static string GetSessionInfo()
		{
			try
			{
				var sessionInfo = Feed.Server.GetSessionInfo();
				var result = FdkVars.RegisterVariable(sessionInfo, "SessionInfo");
				return result;
			}
			catch (Exception ex)
			{
				Log.Error(ex);
				throw;
			}     
        }
        static readonly ILog Log = LogManager.GetLogger(typeof(FdkSessionInfo));

        public static string PlatformCompany(string varName)
        {
            var session = FdkVars.GetValue<SessionInfo>(varName);
            return session.PlatformCompany;
        }
        public static string PlatformName(string varName)
        {
            var session = FdkVars.GetValue<SessionInfo>(varName);
            return session.PlatformName;
        }
        public static string TradingSessionId(string varName)
        {
            var session = FdkVars.GetValue<SessionInfo>(varName);
            return session.TradingSessionId;
        }
        public static DateTime CloseTime(string varName)
        {
            var session = FdkVars.GetValue<SessionInfo>(varName);
            return session.CloseTime;
        }
        public static DateTime EndTime(string varName)
        {
            var session = FdkVars.GetValue<SessionInfo>(varName);
            return session.EndTime;
        }
        public static DateTime OpenTime(string varName)
        {
            var session = FdkVars.GetValue<SessionInfo>(varName);
            return session.OpenTime;
        }
        public static DateTime StartTime(string varName)
        {
            var session = FdkVars.GetValue<SessionInfo>(varName);
            return session.StartTime;
        }
        public static bool IsClosed(string varName)
        {
            var session = FdkVars.GetValue<SessionInfo>(varName);
            return session.IsClosed;
        }
        public static int ServerTimeZoneOffset(string varName)
        {
            var session = FdkVars.GetValue<SessionInfo>(varName);
            return session.ServerTimeZoneOffset;
        }
        public static string Status(string varName)
        {
            var session = FdkVars.GetValue<SessionInfo>(varName);
            return session.Status.ToString();
        }
        
    }
}

File: FDK2R/FdkRLib/RPackage/R/FdkSessionInfo.R
#' Gets the session fields
#' 
#' @export
ttSessionInfo <- function(){
  varName = GetSessionInfo()
  
  platformCompany = PlatformCompany(varName)
  platformName = PlatformName(varName)
  tradingSessionId = TradingSessionId(varName)
  closeTime = CloseTime(varName)
  endTime = EndTime(varName)
  openTime = OpenTime(varName)
  startTime = StartTime(varName)
  serverTimeZoneOffset = ServerTimeZoneOffset(varName)
  status = Status(varName)
  
  UnregisterVar(varName)
  data.table(platformCompany, platformName, tradingSessionId, 
	startTime, endTime, openTime, closeTime, 
	serverTimeZoneOffset, status)      
}

# ****
#' Gets the bars as requested
#' 
GetSessionInfo <- function() {
  rClr::clrCallStatic('RHost.FdkSessionInfo', 'GetSessionInfo')
}

PlatformCompany <- function(varName)
{
 rClr::clrCallStatic('RHost.FdkSessionInfo', 'PlatformCompany', varName)
}

PlatformName <- function(varName)
{
 rClr::clrCallStatic('RHost.FdkSessionInfo', 'PlatformName', varName)
}

TradingSessionId <- function(varName)
{
 rClr::clrCallStatic('RHost.FdkSessionInfo', 'TradingSessionId', varName)
}

CloseTime <- function(varName)
{
 rClr::clrCallStatic('RHost.FdkSessionInfo', 'CloseTime', varName)
}

EndTime <- function(varName)
{
 rClr::clrCallStatic('RHost.FdkSessionInfo', 'EndTime', varName)
}

OpenTime <- function(varName)
{
 rClr::clrCallStatic('RHost.FdkSessionInfo', 'OpenTime', varName)
}

StartTime <- function(varName)
{
 rClr::clrCallStatic('RHost.FdkSessionInfo', 'StartTime', varName)
}

IsClosed <- function(varName)
{
 rClr::clrCallStatic('RHost.FdkSessionInfo', 'IsClosed', varName)
}

ServerTimeZoneOffset <- function(varName)
{
 rClr::clrCallStatic('RHost.FdkSessionInfo', 'ServerTimeZoneOffset', varName)
}

Status <- function(varName)
{
 rClr::clrCallStatic('RHost.FdkSessionInfo', 'Status', varName)
}
